test/e2e/framework/kubesystem/kubesystem.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubesystem
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2essh "k8s.io/kubernetes/test/e2e/framework/ssh"
+)
+
+const (
+	// ssh port
+	sshPort = "22"
+)
+
+// RestartControllerManager restarts the kube-controller-manager.
+func RestartControllerManager() error {
+	// TODO: Make it work for all providers and distros.
+	if !framework.ProviderIs("gce", "aws") {
+		return fmt.Errorf("unsupported provider for RestartControllerManager: %s", framework.TestContext.Provider)
+	}
+	if framework.ProviderIs("gce") && !framework.MasterOSDistroIs("gci") {
+		return fmt.Errorf("unsupported master OS distro: %s", framework.TestContext.MasterOSDistro)
+	}
+	cmd := "pidof kube-controller-manager | xargs sudo kill"
+	framework.Logf("Restarting controller-manager via ssh, running: %v", cmd)
+	result, err := e2essh.SSH(cmd, net.JoinHostPort(framework.GetMasterHost(), sshPort), framework.TestContext.Provider)
+	if err != nil || result.Code != 0 {
+		e2essh.LogResult(result)
+		return fmt.Errorf("couldn't restart controller-manager: %v", err)
+	}
+	return nil
+}
+
+// WaitForControllerManagerUp waits for the kube-controller-manager to be up.
+func WaitForControllerManagerUp() error {
```

Comment:
Tests largely rely on `"k8s.io/apimachinery/pkg/util/wait"` for polling ops.

Reply:
I am not sure I understand your comment correctly.
Do you mean we can replace this function with the PollImmediate()?

Reply:
/hold 
Looking forward to  @copejon 's  reply , or others' opinions
/cc @SataQiu 
what do you think?

Reply:
My point being that it would just cut back repeated code and adhere more closely to e2e conventions.  I meant the comment only as a suggestion though - please feel free to disregard.

To add some context to my first comment:
The `wait` package offers a handful of generic functions that will execute a callback at given intervals, with some being cancelable with a context, timeout, or by channel. The `PollImmediate` function for example: 

`func PollImmediate(interval, timeout time.Duration, condition ConditionFunc)`

executes the `condition` callback every `interval`, and can be canceled via the channel.

The only diff between `Poll` and `PollImmediate` is that the former executes the callback after waiting `interval`; the later executes the callback and then waits `interval`.

Reply:
@copejon Thanks for your comment, I got the point.
It is a nice idea for me because e2e tests consists of many waiting methods, these functions can be made simple by using PollImmediate() as you suggested.
I'd like to keep this PR as current one to make it simple as the original scope and will try doing that with different PRs, I guess it requires multiple PRs for replacing many waiting methods.

/hold cancel